attalos/dataset/vg_prep.py

```diff
@@ -0,0 +1,176 @@
+        self.metadata_filename = self.get_candidate_filename(VISUAL_GENOME_METADATA)
+        self.images_filename = self.get_candidate_filename(VISUAL_GENOME_IMAGES)
+        self.objects_filename = self.get_candidate_filename(VISUAL_GENOME_OBJECTS)
+        self.attributes_filename = self.get_candidate_filename(VISUAL_GENOME_ATTRIBUTES)
+        self.download_dataset()
+        self.load_metadata()
+        self.images_file_handle = None
+
+    def download_dataset(self, extract_all_images=False):
+        """
+        Downloads the dataset if it's not already present in the download directory
+        Returns:
+        """
+
+        self.download_if_not_present(self.metadata_filename, VISUAL_GENOME_METADATA)
+        self.download_if_not_present(self.images_filename, VISUAL_GENOME_IMAGES )
+        self.download_if_not_present(self.relationships_filename, VISUAL_GENOME_RELATIONSHIPS)
+        self.download_if_not_present(self.objects_filename, VISUAL_GENOME_OBJECTS)
+        self.download_if_not_present(self.attributes_filename, VISUAL_GENOME_ATTRIBUTES)
+
+        if not os.path.exists(self.metadata_filename[:-4]):
+            zipref = zipfile.ZipFile(self.metadata_filename,'r')
+            zipref.extractall(self.data_dir)
+        if not os.path.exists(self.relationships_filename[:-4]):
+            zipref = zipfile.ZipFile(self.relationships_filename,'r')
+            zipref.extractall(self.data_dir)
+        if not os.path.exists(self.objects_filename[:-4]):
+            zipref = zipfile.ZipFile(self.objects_filename,'r')
+            zipref.extractall(self.data_dir)
+        if not os.path.exists(self.attributes_filename[:-4]):
+            zipref = zipfile.ZipFile(self.attributes_filename,'r')
+            zipref.extractall(self.data_dir)
+
+        if extract_all_images:
+            zipref = zipfile.ZipFile(self.images_filename,'r')
+            zipref.extractall(self.data_dir)
+
+    def load_metadata(self):
+        """
+        Load the VGG dataset to allow for efficient iteration
+        Returns:
+
+        """
+        if self.split == SplitType.TRAIN:
+            split_name = 'train'
+        elif self.split == SplitType.VAL:
+            split_name = 'val'
+        else:
+            raise NotImplementedError('Split type not yet implemented')
+
+        self.item_info = json.loads(open(self.metadata_filename[:-4], 'r').read().decode('ascii'))
+        # self.item_info=lapi.GetAllImageData(dataDir=self.data_dir)
+        
+
+    def get_key(self, key):
+        """
+        Return metadata about key
+        Args:
+            key: ID who's metadata we'd like to extract
+
+        Returns:
+            RecordMetadata: Returns ParserMetadata object containing metadata about item
+        """
+        
+        # item = self.item_info[key]
+        # return RecordMetadata(id=key, images_name=item['fname'], tags=item['tags'], captions=item['captions'])
+
+        return self.item_info[key-1]
```

Comment:
How does the "-1" interact with the list_keys? Is there a record that never gets read?